src/transformStateWithClones.js

```diff
@@ -8,6 +8,31 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const result = [];
+  const tempState = { ...state };
+
+  for (const action of actions) {
+    if (action['type'] === 'addProperties') {
+      Object.keys(action['extraData']).forEach(key => {
```

Comment:
You can use Object.assign(tempState, extraData) and remove a redundant iteration

Reply:
I change this